apps/nextjs-website/src/lib/webinars/webinarQuestions.ts

```diff
@@ -35,64 +33,50 @@ export type WebinarQuestion = {
   readonly highlightedBy?: string;
 };
 
+export type UpdateExpression<T> =
+  | {
+      readonly operation: 'update';
+      readonly value: T;
+    }
+  | {
+      readonly operation: 'remove';
+    };
+
+export type WebinarQuestionUpdate = Pick<
+  WebinarQuestion,
+  'webinarId' | 'createdAt'
+> & {
+  readonly hiddenBy?: UpdateExpression<string>;
+  readonly highlightedBy?: UpdateExpression<string>;
+};
+
 export const insertWebinarQuestion = (question: InsertWebinarQuestion) =>
   pipe(
     R.ask<Pick<WebinarEnv, 'dynamoDBClient' | 'nowDate'>>(),
     R.map(({ dynamoDBClient, nowDate }) => {
       const createdAt = nowDate();
       // create put command
-      const questionItem = makeDynamodbItemFromWebinarQuestion({
-        webinarId: question.webinarId,
-        createdAt: createdAt,
-        question: question.question,
-      });
-
       const putCommand = new PutItemCommand({
         TableName: 'WebinarQuestions',
-        Item: {
-          ...questionItem,
-          highlightedBy: { NULL: true },
-          hiddenBy: { NULL: true },
-        },
+        Item: makeDynamodbItemFromWebinarQuestion({
+          webinarId: question.webinarId,
+          createdAt: createdAt,
+          question: question.question,
+        }),
       });
       return TE.tryCatch(() => dynamoDBClient.send(putCommand), E.toError);
     }),
     RTE.map(() => void 0)
```

Comment:
I'd add a comment explainig the last line is used to discard the result of the computation, because we are interested only in its side effects.
BTW, I think that it would be less elegant but more ts style and readable to the average developer to just get the client and the date and sending the command